codegen/gateway.go

```diff
@@ -101,32 +101,46 @@ func NewClientSpec(jsonFile string, h *PackageHelper) (*ClientSpec, error) {
 		)
 	}
 
-	clientConfigGeneric := map[string]interface{}{}
-	err = json.Unmarshal(bytes, &clientConfigGeneric)
+	classConfig := map[string]interface{}{}
+	err = json.Unmarshal(bytes, &classConfig)
 	if err != nil {
 		return nil, errors.Wrapf(
 			err, "Could not parse json file %s: ", jsonFile,
 		)
 	}
 
-	clientType := clientConfigGeneric["clientType"].(string)
-	if clientType == "http" {
+	className := classConfig["name"].(string)
+	classType := classConfig["type"].(string)
+
+	clientConfig := classConfig["config"].(map[string]interface{})
+
+	if classType == "http" {
 		// A better solution for client configs is needed here.
 		// Assuming a string map is too restrictive
-		clientConfigObj := map[string]string{}
-		err := json.Unmarshal(bytes, &clientConfigObj)
+		httpClientConfig := map[string]string{}
 
+		serializedConfig, err := json.Marshal(clientConfig)
```

Comment:
It is not efficient to marshal and then unmarshal to get the `clientConfig` here.

Define `classConfig` with explicit type info:

` 
type ClassConfig struct {
    Name string
    Type string
    ClientConfig map[string]string
}
`

You can get all info without type assertions